From b00721f77a25d4312c719a377683988e232e3af8 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Fri, 23 Dec 2016 22:07:26 +0500
Subject: [PATCH] tiledPlayback: fix path searching for TileDecoder

This is primarily OE specific but will work in generic
environments as well because we use relative path
searching to achieve this.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 samples/tiledPlayback/build/CMakeLists.txt | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/samples/tiledPlayback/build/CMakeLists.txt b/samples/tiledPlayback/build/CMakeLists.txt
index 52c4e50..8977e4b 100755
--- a/samples/tiledPlayback/build/CMakeLists.txt
+++ b/samples/tiledPlayback/build/CMakeLists.txt
@@ -141,15 +141,11 @@ find_library( VULKAN_LIBRARIES
 )
 mark_as_advanced( VULKAN_LIBRARIES )
 
+if( EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/../../../modules/TileDecoder/lib/${BITNESS_SUFFIX}/${CMAKE_BUILD_TYPE}/libTileDecoder.so )
 
+    set( TILE_DECODER_LIBRARIES ${CMAKE_CURRENT_SOURCE_DIR}/../../../modules/TileDecoder/lib/${BITNESS_SUFFIX}/${CMAKE_BUILD_TYPE}/libTileDecoder.so CACHE FILEPATH "" )
 
-find_library( TILE_DECODER_LIBRARIES
-
-	NAMES TileDecoder
-
-	PATHS ../../../modules/TileDecoder/lib/${BITNESS_SUFFIX}/${CMAKE_BUILD_TYPE} NO_DEFAULT_PATH 
-
-)
+endif()
 
 # Tweaks for cygwin makefile to work with windows-style path
 
@@ -277,15 +273,15 @@ set_target_properties( ${SAMPLE_NAME} PROPERTIES
 
 
 
-if ( NOT TILE_DECODER_LIBRARIES)
+if ( NOT EXISTS ${TILE_DECODER_LIBRARIES} )
 
     target_link_libraries( ${SAMPLE_NAME} LINK_PUBLIC TileDecoder   ${VULKAN_LIBRARIES} dl pthread xcb)
 
-else ( TILE_DECODER_LIBRARIES_FOUND)
+else ( )
 
     target_link_libraries( ${SAMPLE_NAME} ${TILE_DECODER_LIBRARIES}   ${VULKAN_LIBRARIES} dl pthread xcb)
 
-endif( NOT TILE_DECODER_LIBRARIES )
+endif ( )
 
 
 # Set output directory to bin
-- 
1.9.1

